fcomm/src/lib.rs

```diff
@@ -0,0 +1,1319 @@
+            let cont = ptr_evaluation.cont.cont_ptr(s);
+
+            IO::<S1> { expr, env, cont }
+        };
+
+        let output_io = {
+            let expr = ptr_evaluation.expr_out.ptr(s, 0, lang);
+            let env = ptr_evaluation.env_out.ptr(s, 0, lang);
+            let cont = ptr_evaluation.cont_out.cont_ptr(s);
+
+            IO::<S1> { expr, env, cont }
+        };
+
+        Ok((input_io, output_io))
+    }
+
+    pub fn opening_io(&self, s: &mut Store<S1>) -> Result<(IO<S1>, IO<S1>), Error> {
+        assert!(self.claim.is_opening());
+
+        let opening = self.claim.opening().expect("expected opening claim");
+        let output = opening.public_output_expression(s);
+        let input = s.read(&opening.input).expect("could not read input");
+
+        let expression = opening.commitment.fun_application(s, input);
+        let outermost = s.intern_cont_outermost();
+
+        let input_io = IO::<S1> {
+            expr: expression,
+            env: empty_sym_env(s),
+            cont: outermost,
+        };
+
+        let output_io = IO::<S1> {
+            expr: output,
+            env: empty_sym_env(s),
+            cont: s.intern_cont_terminal(),
+        };
+
+        Ok((input_io, output_io))
+    }
+
+    pub fn io(
+        &self,
+        s: &mut Store<S1>,
+        lang: &Lang<S1, Coproc<S1>>,
+    ) -> Result<(IO<S1>, IO<S1>), Error> {
+        match self.claim {
+            Claim::Evaluation(_) => self.evaluation_io(s),
+            Claim::PtrEvaluation(_) => self.ptr_evaluation_io(s, lang),
+            Claim::Opening(_) => self.opening_io(s),
+        }
+    }
+
+    fn io_vecs(&self, lang: &Lang<S1, Coproc<S1>>) -> Result<(Vec<S1>, Vec<S1>), Error> {
+        let s = &mut Store::<S1>::default();
+
+        self.io(s, lang)
+            .map(|(i, o)| (i.to_inputs(s), o.to_inputs(s)))
+    }
+}
+
+impl VerificationResult {
+    fn new(verified: bool) -> Self {
+        Self { verified }
+    }
+}
+
+pub fn evaluate<F: LurkField>(
+    store: &mut Store<F>,
+    expr: Ptr<F>,
+    supplied_env: Option<Ptr<F>>,
+    limit: usize,
+    lang: &Lang<F, Coproc<F>>,
+) -> Result<(IO<F>, usize), Error> {
+    let env = supplied_env.unwrap_or_else(|| empty_sym_env(store));
+    let mut evaluator = Evaluator::new(expr, env, store, limit, lang);
+
+    let (io, iterations, _) = evaluator.eval().map_err(Error::EvaluationFailure)?;
+
+    assert!(<lurk::eval::IO<F> as Evaluable<F, Witness<F>, Coproc<F>>>::is_terminal(&io));
+    Ok((io, iterations))
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use lurk::public_parameters::FileStore;
+    use std::path::Path;
+    use std::sync::Arc;
+    use tempfile::Builder;
+
+    use lurk::eval::lang::{Coproc, Lang};
+    use lurk::proof::{nova::NovaProver, Prover};
+    use lurk::public_parameters::public_params;
+    use lurk::z_data::{from_z_data, to_z_data};
+
+    //#[test]
+    //fn test_cert_serialization() {
```

Comment:
This can be uncommented

Reply:
See the comment below on the `Cert` struct.

Reply:
Please reinstate this uncommented test which you are removing from `public_parameters`:
https://github.com/lurk-lab/lurk-rs/pull/447/files#diff-372f8376bdf97a030ac402c26c94fd3ba3ee64fb207d20c2ff7c446f0a4bd916L1151-L1173